src/rust/ide/src/controller/visualization.rs

```diff
@@ -22,14 +21,30 @@ use std::rc::Rc;
 /// Enumeration of errors used in `Visualization Controller`.
 #[derive(Debug,Fail)]
 #[allow(missing_docs)]
-pub enum VisualizationError {
+pub enum Error {
     #[fail(display = "Visualization \"{}\" not found.", identifier)]
     NotFound {
         identifier : VisualizationPath
     },
+    #[fail(display = "JavaScript visualization \"{}\" failed to be prepared.", identifier)]
+    Preparation {
+        identifier : VisualizationPath,
+        #[cause]
+        cause      : failure::Error,
```

Comment:
I would not align `:` in such a case. It looks strange. Only ni lines that are gluied together imo